### DISCUSSION TITLE
Clearing the store

### DISCUSSION THREAD

**Discussion by ericbroberic** [alpinejs/alpine#3652] [1. Help] [answered]
Hi!

I'm new to alpinejs and I've been looking through the documentation and the source code. I can't seem to find a way to clear the store once you set a value. Looks like you can set it to `null` if you know the `name`, but I can't find a way to do a full clear, where the `name` is not known ahead of time. Is there anyway to do this? 

Thanks!

**Comment by ekwoka** [comment, marked as answer]
No.

I also don't know a context in which this would ever be something you need....

**Reply by ericbroberic** [reply]
Really? The store is essentially a map where I would expect standard operations such as `clear`, `remove`, `get`, and `set`. Clearing and removing would definitely be useful in the case that you need to remove data for cleaning up memory, especially considering the logic to implement would be extremely simple. My specific case is that I'm filling out the `store` through an extended attribute and wanted to clear the data before storing the new attribute data whenever it changes. I'm not using the `store` anywhere else, so I was just going to clear it for convenience sake. However, I can easily workaround it by just getting the old attribute data and setting all the `store` values to `null`.

I appreciate the quick response, and thank you for your answer!

**Reply by ekwoka** [reply]
or just set the whole thing to null

**Reply by SimoTod** [reply]
> Really? The store is essentially a map where I would expect standard operations such as `clear`, `remove`, `get`, and `set`. Clearing and removing would definitely be useful in the case that you need to remove data for cleaning up memory, especially considering the logic to implement would be extremely simple. My specific case is that I'm filling out the `store` through an extended attribute and wanted to clear the data before storing the new attribute data whenever it changes. I'm not using the `store` anywhere else, so I was just going to clear it for convenience sake. However, I can easily workaround it by just getting the old attribute data and setting all the `store` values to `null`.
> 
> I appreciate the quick response, and thank you for your answer!

The store is more like a global state, not really a map. It doesn't persist between pages so you generally don't need to empty it. Resetting a store is not a common use case (it's the same as resetting an x-data) so if you have memory issues on a single page because of a store, you can take care of it by setting problematic keys to null as suggested.

**Reply by ericbroberic** [reply]
@SimoTod I think there maybe a bug with setting keys to null. It seems to throw an error in `interceptor.js`.

https://codepen.io/Eric-Benn/pen/poQaNjr

**Reply by SimoTod** [reply]
It looks like a bug. It works with normal stores ~or using undefined~

```js
Alpine.store('test', 5);
console.log(Alpine.store('test'));

Alpine.store('test3', {value: null});
console.log(Alpine.store('test3')); 
```
